libsolidity/formal/BMC.cpp

```diff
@@ -184,7 +185,42 @@ bool BMC::visit(IfStatement const& _node)
 		);
 	m_context.popSolver();
 
-	SMTEncoder::visit(_node);
+	auto mergePathConditions = [](smtutil::Expression const& trueCondition, smtutil::Expression const& falseCondition) {
+		return trueCondition || falseCondition;
+	};
+
```

Comment:
In general, could we instead add the code below to `SMTEncoder::visitBranch` and `SMTEncoder::visit(IfStatement)`?

Reply:
I mean, we could also keep it here in BMC, but then we should delete the stuff in SMTEncoder since I think no one is using the If handling there.